### ISSUE TITLE
This repository shows "No History"

### ISSUE THREAD

**Issue by cg31** [huizhougit/githd#89]
After cloning this repo and open the whole repo in vscode:
https://github.com/PacktPublishing/Modern-Time-Series-Forecasting-with-Python

If I select "GitHD: View Foler History" at top-level (whole repo), I will get a "No History".

If I cleick a folder then select "GitHD: View Foler History", the history will show normally.

It is kind wierd.

**Comment by huizhougit** [comment]
Can you share a screen shot how did you select it at the top level?

**Comment by cg31** [comment]
First open that repo, then

1 Righ-click on the empty space to evoke the menu
2 Select View Folder History
3 The history view should display the repo history

It works for almost all repos except this one

![git-history](https://github.com/user-attachments/assets/81e6c74f-b58f-438a-bdf7-a45c918efe6e)

**Comment by huizhougit** [comment]
I cannot repro it. Can you please share the log?
Go to vscode settings and search "githd", change the trace level to verbose. Then go to the `output` panel and select `githd` 
![image](https://github.com/user-attachments/assets/67645011-c65f-4128-af81-2a352dfff9a0)

**Comment by cg31** [comment]
I found a workaround: after "No History" is showed, I click the "GitHD: View History" icon on top-right of the view, the history log will show normally. It seems something is not updated.

I have two verbose logs now, one is for the issue, another is after clicking the icon.

1 after selecting the "view history" in menu, "No History" will show
```
[18:11:53.961][Verbose] Command: githd.viewFileHistory
[18:11:53.962][Info] Model: set historyViewContext - {"specifiedPath":{"$mid":1,"fsPath":"~/test","path":"~/test","scheme":"file"},"repo":{"root":"~/test/","remoteUrl":"https://github.com/PacktPublishing/Modern-Time-Series-Forecasting-with-Python-2E"},"branch":"main"}
[18:11:53.962][Verbose] HistoryView: onDidChangeHistoryViewContext
[18:11:53.962][Verbose] HistoryView: updating canceled
[18:11:53.972][Verbose] HistoryView: _startLoading
[18:11:53.972][Verbose] HistoryView: _reset
[18:11:53.972][Info] HistoryView: _update, content size 1, left count 0
[18:11:53.972][Verbose] HistoryView: provideTextDocumentContent length: 1
[18:11:53.972][Verbose] HistoryView: _moveToTop
[18:11:54.326][Verbose] HistoryView: onDidChangeTextDocument
[18:11:54.326][Verbose] HistoryView: _resetDecorations
[18:11:54.326][Verbose] HistoryView: _setDecorations content length: 1, _subjectDecorationOptions size: 0
[18:11:54.326][Verbose] HistoryView: _loadingDecoration used
[18:11:54.326][Info] HistoryView: _updateContent. {"branch":"main","repo":{"root":"~/test/","remoteUrl":"https://github.com/PacktPublishing/Modern-Time-Series-Forecasting-with-Python-2E"},"specifiedPath":{"$mid":1,"fsPath":"~/test","path":"~/test","scheme":"file"}}
[18:11:54.327][Verbose] HistoryView: left count 31, current total count 61, loading count 90, load more false
[18:11:54.328][Info] HistoryView: _update, content size 8092, left count 31
[18:11:54.328][Verbose] HistoryView: provideTextDocumentContent length: 8092
[18:11:54.417][Verbose] HistoryView: onDidChangeTextDocument
[18:11:54.417][Verbose] HistoryView: _setDecorations content length: 8092, _subjectDecorationOptions size: 339
[18:11:54.418][Info] HistoryView: _updateContent. {"branch":"main","repo":{"root":"~/test/","remoteUrl":"https://github.com/PacktPublishing/Modern-Time-Series-Forecasting-with-Python-2E"},"specifiedPath":{"$mid":1,"fsPath":"~/test","path":"~/test","scheme":"file"}}
[18:11:54.418][Verbose] HistoryView: left count 0, current total count 61, loading count 31, load more false
[18:11:54.418][Info] HistoryView: _update, content size 10, left count 0
[18:11:54.418][Verbose] HistoryView: provideTextDocumentContent length: 10
[18:11:54.551][Verbose] HistoryView: onDidChangeTextDocument
[18:11:54.551][Verbose] HistoryView: _setDecorations content length: 10, _subjectDecorationOptions size: 339
[18:11:54.552][Verbose] PanelViewProvider: update: commits 0 buckets 91
```
2 after clicking the icon, the normal history will be displayed
```
[18:13:51.997][Verbose] Command: githd.viewHistory
[18:13:51.997][Info] Model: set historyViewContext - {"repo":{"root":"~/test/","remoteUrl":"https://github.com/PacktPublishing/Modern-Time-Series-Forecasting-with-Python-2E"},"branch":"main"}
[18:13:51.997][Verbose] HistoryView: onDidChangeHistoryViewContext
[18:13:51.998][Verbose] HistoryView: updating canceled
[18:13:52.008][Verbose] HistoryView: _startLoading
[18:13:52.008][Verbose] HistoryView: _reset
[18:13:52.008][Info] HistoryView: _update, content size 1, left count 0
[18:13:52.008][Verbose] HistoryView: provideTextDocumentContent length: 1
[18:13:52.008][Verbose] HistoryView: _moveToTop
[18:13:52.018][Verbose] HistoryView: onDidChangeTextDocument
[18:13:52.018][Verbose] HistoryView: _resetDecorations
[18:13:52.019][Verbose] HistoryView: _setDecorations content length: 1, _subjectDecorationOptions size: 0
[18:13:52.019][Verbose] HistoryView: _loadingDecoration used
[18:13:52.019][Info] HistoryView: _updateContent. {"branch":"main","repo":{"root":"~/test/","remoteUrl":"https://github.com/PacktPublishing/Modern-Time-Series-Forecasting-with-Python-2E"}}
[18:13:52.019][Verbose] HistoryView: left count 65, current total count 95, loading count 90, load more false
[18:13:52.021][Info] HistoryView: _update, content size 12232, left count 65
[18:13:52.021][Verbose] HistoryView: provideTextDocumentContent length: 12232
[18:13:52.182][Verbose] HistoryView: onDidChangeTextDocument
[18:13:52.182][Verbose] HistoryView: _setDecorations content length: 12232, _subjectDecorationOptions size: 503
[18:13:52.182][Info] HistoryView: _updateContent. {"branch":"main","repo":{"root":"~/test/","remoteUrl":"https://github.com/PacktPublishing/Modern-Time-Series-Forecasting-with-Python-2E"}}
[18:13:52.182][Verbose] HistoryView: left count 0, current total count 95, loading count 65, load more false
[18:13:52.183][Info] HistoryView: _update, content size 12940, left count 0
[18:13:52.183][Verbose] HistoryView: provideTextDocumentContent length: 12940
[18:13:52.208][Verbose] HistoryView: onDidChangeTextDocument
[18:13:52.208][Verbose] HistoryView: _setDecorations content length: 12940, _subjectDecorationOptions size: 537
[18:13:52.209][Verbose] PanelViewProvider: update: commits 52 buckets 91
```

**Comment by huizhougit** [comment]
Thanks for the log. I know the issue now and will fix it soon. The workaround is turn off the express mode. BTW, the stats chart feature will only work when express mode is turned off.

**Comment by huizhougit** [comment]
Fixed by https://github.com/huizhougit/githd/commit/67ca850e7374c1f35505285a0b024ec6bcc040cd

**Comment by huizhougit** [comment]
The fix is in v2.5.3. Please let me know if you still hit it.

**Comment by cg31** [comment]
Great. It is indeed fixed.
